doc/tutorials/enable_ivshmem.rst

```diff
@@ -0,0 +1,151 @@
+``ivshmem`` hv-land through ACRN XML configuration(:ref:`ACRN
+Configuration Tool <acrn_configuration_tool>`) before using it.
+
+- Enable ``ivshmem`` hv-land in ACRN XML configuration file. For example, the
+  hybrid_rt scenario and whl-ipc-i5 board XML configuration file is located in
+  ``$PWD/misc/vm_configs/xmls/config-xmls/whl-ipc-i5/hybrid_rt.xml``
+
+   - Configure ``IVSHMEM_ENABLED`` to ``y`` in ACRN scenario XML configuration
+     to enable ``ivshmem`` hv-land
+
+   - Configure ``IVSHMEM_REGION`` to specify the shared memory name, size and
+     communication VMs in ACRN scenario XML configuration. The ``IVSHMEM_REGION``
+     format such as ``shm_name,shm_size,VM IDs``:
+
+	-  ``shm_name`` - Specify a shared memory name. The name needs to start
+	   with the ``hv:/`` prefix. For example, ``hv:/shm_region_0``
+
+	-  ``shm_size`` - Specify a shared memory size
+
+	-  ``VM IDs``   - Specify the VM IDs to use the same shared memory
+	   communication and separate it with ``:``. For example, the
+	   communication between VM0 and VM2, it can be written as ``0:2``
+
+- Build the XML configuration, refer to :ref:`Build ACRN from Source <getting-started-building>`
+
+Inter-VM Communication Example
+******************************
+
+**The following example uses inter-vm communication(dm-land) between two
+Linux-based post-launched VMs (VM1 and VM2)**
+
+.. note:: An ``ivshmem`` Windows driver exists and can be found `here <https://github.com/virtio-win/kvm-guest-drivers-windows/tree/master/ivshmem>`_
+
+1. Add a new virtual PCI device for both VMs: the device type is
+   ``ivshmem``, shared memory name is ``sos:/test``, and shared memory size is
+   2097152(2M) bytes. Both VMs must have the same shared memory name and size:
+
+   - VM1 Launch Script Sample
+
+     .. code-block:: none
+        :emphasize-lines: 7
+
+        acrn-dm -A -m $mem_size -s 0:0,hostbridge \
+         -s 2,pci-gvt -G "$2" \
+         -s 5,virtio-console,@stdio:stdio_port \
+         -s 6,virtio-hyper_dmabuf \
+         -s 3,virtio-blk,/home/acrn/uos1.img \
+         -s 4,virtio-net,tap0 \
+         -s 6,ivshmem,sos:/test,2097152\
+         -s 7,virtio-rnd \
+         --ovmf /usr/share/acrn/bios/OVMF.fd \
+         $vm_name
+
+
+   - VM2 Launch Script Sample
+
+     .. code-block:: none
+        :emphasize-lines: 5
+
+        acrn-dm -A -m $mem_size -s 0:0,hostbridge \
+         -s 2,pci-gvt -G "$2" \
+         -s 3,virtio-blk,/home/acrn/uos2.img \
+         -s 4,virtio-net,tap0 \
+         -s 5,ivshmem,sos:/test,2097152\
+         --ovmf /usr/share/acrn/bios/OVMF.fd \
+         $vm_name
+
+2. Boot two VMs and use ``lspci | grep "shared memory"`` to verify that the virtual device is ready for each VM.
+
+   -  For VM1, it shows ``00:06.0 RAM memory: Red Hat, Inc. Inter-VM shared memory (rev 01)``
+   -  For VM2, it shows ``00:05.0 RAM memory: Red Hat, Inc. Inter-VM shared memory (rev 01)``
+
+3. Use these commands to probe the device::
```

Comment:
Can you specify if these commands are to be used in *all* VMs that want to use the shared memory? I'm assuming so but let's make it clear.

Reply:
Yes， I will add a note here that these commands are applicable to Linux-based guests with CONFIG_UIO and CONFIG_UIO_PCI_GENERIC enabled